src/openfermion/utils/_bch_expansion.py

```diff
@@ -16,11 +16,50 @@
 from scipy.misc import comb, factorial
 
 
-def bch_expand(x, y, order=6):
-    """Compute log[e^x e^y] using the Baker-Campbell-Hausdorff formula.
+def bch_expand(*ops, **kwargs):
+    """Compute log[e^{x_1} ... e^{x_N}] using the BCH formula.
 
     This implementation is explained in arXiv:1712.01348.
 
+    Args:
+        ops: A sequence of operators of the same type for which multiplication
+            and addition are supported. For instance, QubitOperators,
+            FermionOperators, or Scipy sparse matrices.
+        keyword arguments:
+            order(int): The max degree of monomial with respect to X and Y
+                to truncate the BCH expansions. Defaults to 6
+
+    Returns:
+        The truncated BCH operator.
+
+    Raises:
+        ValueError: invalid order parameter.
+        TypeError: operator types are not all the same.
+    """
+    order = kwargs.get('order', 6)
+    if (not isinstance(order, int)) or order < 0:
+        raise ValueError('Invalid order parameter.')
+    if len(set(type(op) for op in ops)) != 1:
+        raise ValueError('Operators must all be of the same type.')
```

Comment:
This should be a `TypeError` (saw you changed it in the docs!). The other thing is the message would be confusing if someone tried `bch_expand()`, I think TypeError is still appropriate there but the message should be different.